### DISCUSSION TITLE
Open Telemetry Support

### DISCUSSION THREAD

**Discussion by JoeHillyard** [TypedProject/ts-express-decorators#1951] [Ideas]
Hi team,

I'm just wondering if there would be any consideration for supporting the OpenTelemetry spec within Ts.ED?

It's still a little early in maturity but I'm interested to know if it would be considered for the future roadmap.

Thanks!

**Comment by Romakita** [comment]
Hello @JoeHillyard 

thanks for this suggestion ;)

Why openTelemetry could be interesting for the framework? it needs a specific integration in Ts.ED?
Is it interesting for any project?

See you

**Comment by JoeHillyard** [comment]
For context, we currently have production systems using routing-controllers and after evaluating frameworks I want to move to Ts.ED - one of our main focuses alongside developer experience is [Cloud Native](https://www.cncf.io/) and OpenTelemetry focuses on Tracing, Metrics & Logging.

I am still learning the stack but I would imagine that tsed/logger would be extended to export OpenTelemetry logs and separate metrics and tracing modules would be required - some APIs/SDKs are stable but some are still being drafted/discussed so it would be a long term strategy :)
